pkg/cmd/server/origin/leaderelection.go

```diff
@@ -81,10 +87,6 @@ func NewLeaderElection(options configapi.MasterConfig, leader componentconfig.Le
 
 	// use the endpoints leader election path.
 	plug := plug.New(false)
-	events := record.NewBroadcaster()
-	events.StartLogging(glog.Infof)
-	events.StartRecordingToSink(&v1core.EventSinkImpl{Interface: v1core.New(kc.Core().RESTClient()).Events("")})
-	lock.LockConfig.EventRecorder = events.NewRecorder(kapi.Scheme, v1corev1.EventSource{Component: name})
 	elector, err := leaderelection.NewLeaderElector(leaderelection.LeaderElectionConfig{
```

Comment:
The legacy code path needs the lock interface as well.  We can't refactor this until we remove the legacy lock in 3.8.